src/components/views/elements/AppPermission.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import url from 'url';
+
+export default class AppPermission extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const curlBase = this.getCurlBase();
+        this.state = {
+            curlBase: curlBase,
```

Comment:
There's no style guideline for this. There's a number of reasons why I prefer it inline:
 - It reduces the number of variables in play in this function by 1.
 - It is more maintainable against modification as there is literally no way `curlBase` can be modified between the function call to `getCurlBase()` and it being set on `this.state`.
 - It's more maintainable against additions as I just need to add a single key/value pair to `this.state`. In the not-inlined form I would need to make another temporary variable and then do the set, which gets messy as the number of additions increases.

Furthermore, the following are not true (if any of these are true, I would not be asking to inline this):
 - The function call has a lot of args.
 - It's not reasonably obvious what the function will do and whether it will have side-effects. In this case it's just a getter so it's obvious.
 - There isn't a nice 1:1 mapping between return arg and `this.state` value.